File: solutions/0070-climbing-stairs/solution.java
class Solution {
    HashMap<Integer, Integer> cache = new HashMap<>();
    public int climbStairs(int n) 
    {
        cache.put(1,1);
        cache.put(2,2);
        return dp(n);
    }
    public int dp(int n)
    {
        if(cache.containsKey(n))
            return cache.get(n);
        else
        {
            cache.put(n, dp(n-1)+dp(n-2));
            return cache.get(n);
        }
    }
}

File: solutions/0070-climbing-stairs/solution.py
class Solution:
    def climbStairs(self, n: int) -> int:
        ans = []
        ans.append(1)
        ans.append(2)
        for i in range(2,n):
            ans.append(ans[i-1] + ans[i-2])
        
        return ans[n-1]
